changelog/_unreleased/2024-03-27-generate-new-token-when-context-not-found.md

```diff
@@ -0,0 +1,11 @@
+---
+title: generate new token when context not found
+issue: NEXT-00000
+author: Jasper Peeters
+author_email: jasper.peeters@meteor.be
+author_github: JasperP98
+---
+
+# Core
+
+* When you call the `\Shopware\Core\System\SalesChannel\Context\SalesChannelContextService::get` and the context is not found, there will be a context created in memory with the given sales channel context token. You would expect a new token to be generated in this case.
```

Comment:
Hey, 

I think you are right with your assumption, but i guess it will break a lot of implementation.

Right now you can just generate a random token in your application before requesting our api and the token stays like you generated it.

Therefore i have to decline this change :/

Reply:
Hi @OliverSkroblin,

I understand that this is a breaking change but would it be a possibility to apply this change for the next major release?